source/IAccessibleHandler/__init__.py

```diff
@@ -798,7 +798,15 @@ def _get_role(self):
 		return controlTypes.Role.PANE
 
 	def event_gainFocus(self):
-		super(SecureDesktopNVDAObject, self).event_gainFocus()
+		from speech.speech import cancelSpeech
+		# NVDA announces the secure desktop when handling the gainFocus event.
+		# Before announcing the secure desktop and entering sleep mode,
+		# cancel speech so that speech does not overlap with the new instance of NVDA
+		# started on the secure desktop.
+		# Cancelling speech was previously handled by a foreground event on SecureDesktopNVDAObject.

```

Comment:
```suggestion
		# Cancelling speech was previously handled by a foreground event fired when focusing SecureDesktopNVDAObject.
		# The foreground event would incorrectly fire on the foreground window of the user desktop.
```

Reply:
@feerrenrut - what do you think about this?

Reply:
How about adding: "and was removed due to the security concerns of setting the foreground object to an object 'below the lock screen'."